app/server/utils/ussd/kenya_ussd_state_machine.py

```diff
@@ -116,6 +116,21 @@ def new_pins_match(self, user_input):
             pins_match = True
         return pins_match
 
+    def authorize_pin(self, pin):
+        authorized = False
+        if self.user.failed_pin_attempts == 3:
+            return False
+        if self.user.failed_pin_attempts < 3:
+            authorized = self.user.check_salt_hashed_password(pin, self.user.pin)
```

Comment:
should have a method on User similar to `verify_password` instead of doing the `check_salt_hashed_password` here. below also

Reply:
I saw the method, but felt like what I was doing wasn't exactly checking a password  since the hashed value was a pin. So that's why I went with hashing and checking the hash. 
Added the two methods to complement each other.

Reply:
perhaps we sidebar some time for discussion about this on the call? auth is one of those sufficiently critical things that it warrents proper discussion lol

Reply:
Though something like verify pin would make sense. I'll look at this again.